dask/dataframe/csv.py

```diff
@@ -0,0 +1,142 @@
+from __future__ import print_function, division, absolute_import
+
+from io import BytesIO
+from warnings import warn
+
+import numpy as np
+import pandas as pd
+
+from ..delayed import delayed
+from .io import from_delayed
+
+from ..bytes import read_bytes
+from ..bytes.compression import seekable_files, files as cfiles
+from ..utils import ensure_bytes
+
+
+delayed = delayed(pure=True)
+
+
+def bytes_read_csv(b, header, kwargs, dtypes=None, columns=None):
```

Comment:
Feather could be perfect for the pieces of a dask.DataFrame, no?
Parquet does have byte blocks amenable to being passed to a single function like with avro, except that the metadata is stored at the foot not head (and the blocks might be in different files). All moot until we have a reader/writer...